tests/expressions/tests.py

```diff
@@ -1043,6 +1043,48 @@ def test_duration_with_datetime(self):
         ).order_by('name')
         self.assertQuerysetEqual(over_estimate, ['e3', 'e4'], lambda e: e.name)
 
+    @skipUnlessDBFeature('supports_microsecond_precision')
+    def test_negative_timedelta_with_microsecond_support_update(self):
+        # subtract 30 microseconds
+        Experiment.objects.update(start=F('start') + datetime.timedelta(microseconds=-30))
+        e0 = Experiment.objects.get(name='e0')
+        self.assertEqual(e0.start, datetime.datetime(2010, 6, 25, 12, 15, 30, 746970))
+        # subtract 30 seconds
+        Experiment.objects.update(start=F('start') + datetime.timedelta(seconds=-30))
+        e0 = Experiment.objects.get(name='e0')
+        self.assertEqual(e0.start, datetime.datetime(2010, 6, 25, 12, 15, 0, 746970))
+        # subtract 30 minutes
+        Experiment.objects.update(start=F('start') + datetime.timedelta(minutes=-30))
+        e0 = Experiment.objects.get(name='e0')
+        self.assertEqual(e0.start, datetime.datetime(2010, 6, 25, 11, 45, 0, 746970))
+        # subtract 2 hours
+        Experiment.objects.update(start=F('start') + datetime.timedelta(hours=-2))
+        e0 = Experiment.objects.get(name='e0')
+        self.assertEqual(e0.start, datetime.datetime(2010, 6, 25, 9, 45, 0, 746970))
+        # subtract 2 days
+        Experiment.objects.update(start=F('start') + datetime.timedelta(days=-2))
+        e0 = Experiment.objects.get(name='e0')
+        self.assertEqual(e0.start, datetime.datetime(2010, 6, 23, 9, 45, 0, 746970))
```

Comment:
Whether or not you use `refresh_from_db()`, you may want to write the test to do less than 10 database roundtrips. It isn't a biggie, but these things add up, especially with Oracle.